src/main.rs

```diff
@@ -142,42 +142,44 @@ fn main() {
         let pass = matches.value_of("PASSWORD").unwrap();
 
         let pkcs12 = load_pkcs12_from_file(pkcs12_file, pass).expect("error reading pkcs12");
-        let acceptor = TlsAcceptor::builder(pkcs12).build().unwrap();
-        let tokio_acceptor = tokio_tls::TlsAcceptor::from(acceptor);
+        let acceptor = native_tls::TlsAcceptor::builder(pkcs12).build().unwrap();
+        let tokio_acceptor = tokio_native_tls::TlsAcceptor::from(acceptor);
         Some(Arc::new(tokio_acceptor))
     } else {
         None
     };
 
-    let mut l = Core::new().unwrap();
-    let handle = l.handle();
-
-    let socket = TcpListener::bind(&addr, &handle).unwrap();
+    let mut socket = Box::pin(TcpListener::bind(&addr).await.unwrap());
 
     info!(log, "Started listening"; "addr" => bind_addr, "tls" => tls);
 
     // This is the main loop where we accept incoming *TCP* connections.
-    let done = socket.incoming().for_each(move |(socket, addr)| {
+    while let Some(Ok(tcp_stream)) = socket.next().await {
+        let addr = if let Ok(addr) = tcp_stream.peer_addr() {
+            addr
+        } else {
+            // TODO: log this
+            continue;
+        };
+
         let peer_log = log.new(o!("ip" => format!("{}", addr.ip()), "port" => addr.port()));
 
         let ctx = ConnectionContext {
             logger: Arc::new(peer_log),
             peer_addr: addr,
         };
 
-        let new_handle = handle.clone();
+        //let new_handle = handle.clone();
 
         let cloned_ctx = ctx.clone();
 
         // Set up a new task for the connection. We do this early so that the logging is correct.
-        let setup_future = futures::lazy(move || {
+        let setup_future = futures::future::lazy(move |_cx: &mut Context| {
```

Comment:
Would be good for readability to shorten `futures::future` into just `future` by `use`ing `futures::future` at the start of the file.